pkg/storage/utils/eosfs/eosfs.go

```diff
@@ -1457,23 +1457,13 @@ func getResourceType(isDir bool) provider.ResourceType {
 }
 
 func (fs *eosfs) extractUIDAndGID(u *userpb.User) (string, string, error) {
-	var uid, gid string
-	if u.Opaque != nil && u.Opaque.Map != nil {
-		if uidObj, ok := u.Opaque.Map["uid"]; ok {
-			if uidObj.Decoder == "plain" {
-				uid = string(uidObj.Value)
-			}
-		}
-		if gidObj, ok := u.Opaque.Map["gid"]; ok {
-			if gidObj.Decoder == "plain" {
-				gid = string(gidObj.Value)
-			}
-		}
+	if u.UidNumber == 0 {
+		return "", "", errors.New("eos: uid missing for user")
 	}
-	if uid == "" || gid == "" {
-		return "", "", errors.New("eos: uid or gid missing for user")
+	if u.GidNumber == 0 {
```

Comment:
We need to make sure that UID = 0 or GID = 0 in user requests is blocked eosfs

Reply:
I'm not sure i understand, if you mean that the statement should be `if u.UidNumber == 0 || u.GidNumber == 0`, then you will notice that i separated it into two if statements to provide a more specific error message.